base/gfspapp/upload_server.go

```diff
@@ -129,10 +131,14 @@ func (g *GfSpBaseApp) GfSpUploadObject(stream gfspserver.GfSpUploadService_GfSpU
 	select {
 	case <-ctx.Done():
 		return nil
-	case <-initCh:
+	case <-initChan:
 		log.CtxDebug(ctx, "received first upload stream data")
-	case <-errChan:
-		return err
+	case err1, ok := <-errChan:
+		if !ok {
+			log.Error("error channel has been closed")
+			return fmt.Errorf("closed error channel")
```

Comment:
It is very likely to return error here even the go routine end properly.

Reply:
fixed